File: 0027-remove-element/0027-remove-element.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int [] newnums = nums.clone(); 
        
        int k =0;
        for (int i=0; i < newnums.length; ++i){
            if(newnums[i]!= val){
                nums[k] = nums[i];
                ++k;
            }
        }

        return k;
    }
}

File: 0027-remove-element/0027-remove-element.cs
public class Solution {
    public int RemoveElement(int[] nums, int val) {
        int n = nums.Length;
        int slow = n -1;
        int fast = n -1;

        while(fast >= 0){
            if(nums[fast] == val){
                nums[fast] = nums[slow];
                nums[slow] = val;
                --slow;
            }
            --fast;
        }
        return slow+1;
    }
}